### PR TITLE
Add missing RSpec linter

### PR DESCRIPTION
**Pull request by kevindew**
Trello: https://trello.com/c/QrxjshEm/270-move-ruby-gems-from-jenkins-to-github-actions

This follows on from https://github.com/alphagov/content-tagger/pull/1389 in adding groundwork before importing [govuk_taxonomy_helpers](https://github.com/alphagov/govuk_taxonomy_helpers)) into this app. 

This adds in linting for RSpec and Rake, but the Rake one is of no consequence. The RSpec linter was a bit of a pain to fix as there were many violations. 

This is definitely best reviewed commit by commit, you may also want to turn off whitespace checking for [Resolve method names in top level RSpec describes](https://github.com/alphagov/content-tagger/commit/d1bc4e203e0db6f9a926ed646c0f59b1682c5b1b) as it's mostly indenting all the existing code for a new `describe` block.

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: spec/models/version_spec.rb [hunk 1]
```diff
@@ -1,54 +1,56 @@
-RSpec.describe Version, ".history" do
-  it "returns the the version history in descending order" do
-    content_id = SecureRandom.uuid
+RSpec.describe Version do
+  describe ".version" do
```
Comment (baisa) [inline spec/models/version_spec.rb:2]: Should this be `describe ".history"`?

File: spec/workers/publish_links_worker_spec.rb [hunk 2]
```diff
@@ -8,8 +8,14 @@
...
     context "with valid links" do
       before do
-        allow_any_instance_of(BulkTagging::TagMapping).to receive(:valid?)
-          .and_return(true)
+        mapping_double = instance_double(
+          BulkTagging::TagMapping,
+          valid?: true,
+          errors: [],
+          mark_as_errored: nil,
+          mark_as_tagged: nil,
```
Comment (baisa) [inline spec/workers/publish_links_worker_spec.rb:16]: Wondering why we adding `mark_as_errored` and ` mark_as_tagged` here?

File: spec/forms/new_project_form_spec.rb [hunk 3]
```diff
@@ -42,7 +42,9 @@
     end
 
     it "returns false with an error added when the CSV fails to parse" do
-      allow_any_instance_of(RemoteCsv)
```
Comment (baisa) [inline spec/forms/new_project_form_spec.rb]: Do we know why `allow_any_instance_of` is not recommended? it seems cleaner than stubbing 'new'?

File: spec/models/version_spec.rb (hunk 1 above)
Comment (kevindew) [reply spec/models/version_spec.rb:2]: Ah yes good spot!

File: spec/workers/publish_links_worker_spec.rb (hunk 2 above)
Comment (kevindew) [reply spec/workers/publish_links_worker_spec.rb:16]: These are methods that the mock object needs in it's usage.

File: spec/forms/new_project_form_spec.rb (hunk 3 above)
Comment (kevindew) [reply spec/forms/new_project_form_spec.rb]: Sure, there's a description on RSpec Style Guide: https://rspec.rubystyle.guide/#any_instance_of

`allow_any_instance` is quite controversial in the RSpec world, with [RSpec themselves](https://relishapp.com/rspec/rspec-mocks/docs/working-with-legacy-code/any-instance) advising against using it unless dealing with legacy code.

I've dealt with the warnings in the more bone headed way here of just fixing the tests like for like, so I imagine a more thorough refactor could do nicer things with them.

**Review by baisa — APPROVED** [review]

**Pull request merged**